src/tinykeys.ts

```diff
@@ -88,6 +88,11 @@ function parse(str: string): KeyBindingPress[] {
  * partially or exactly.
  */
 function match(event: KeyboardEvent, press: KeyBindingPress): boolean {
+	// match special characters like '?' and '!'
+	if (/^[^A-Za-z0-9]$/.test(event.key) && press[1] === event.key) {
+		return true
+	}
```

Comment:
I'm confused why this isn't handled by:

```js
(
  press[1].toUpperCase() !== event.key.toUpperCase() &&
  press[1] !== event.code
) ||
```

```js
'?'.toUpperCase() === '?' // true
```

Should we also be comparing:

```js
(
  press[1] !== event.key &&
  press[1].toUpperCase() !== event.key.toUpperCase() &&
  press[1] !== event.code
) ||
```

Reply:
I might be missing something super obvious here but this case:

```
( press[1].toUpperCase() !== event.key.toUpperCase() && press[1] !== event.code ) ||
```

is correct, it's properly returning `false`. However because it returns `false`, we check the second value, and then third value, which:

```
KEYBINDING_MODIFIER_KEYS.find(mod => {
  return !press[0].includes(mod) && press[1] !== mod && getModifierState(event, mod)
})
```

returns `Shift`, which when we apply `!"Shift"` it returns `false` 😅